src/Illuminate/Collections/LazyCollection.php

```diff
@@ -1074,6 +1074,30 @@ public function sole($key = null, $operator = null, $value = null)
             ->sole();
     }
 
+    /**
+     * Get the first item in the collection, but if no items exist throw an exception.
+     *
+     * @param  mixed  $key
+     * @param  mixed  $operator
+     * @param  mixed  $value
+     * @return mixed
+     *
+     * @throws \Illuminate\Collections\ItemNotFoundException
+     */
+    public function firstOrFail($key = null, $operator = null, $value = null)
+    {
+        $filter = func_num_args() > 1
+            ? $this->operatorForWhere(...func_get_args())
+            : $key;
+
+        return $this
+            ->when($filter)
+            ->filter($filter)
+            ->take(2)
```

Comment:
If I'm not mistaken, the ```take(2)``` is the logic from ```sole``` call. You can do ```take(1)``` because it shouldn't matter to check for more than 1 item in collection.

Reply:
Good spot thanks